articles/key-design-of-hdfs-architecture/index.md

```diff
@@ -0,0 +1,120 @@
+
+Big data framework is characterized by 4Vs namely; Variety (data is of various forms and types), Velocity (data processing speed is high), Value (low data value density), and Volume (massive amount of data).
+
+Among the frameworks that can do the processing of data with the above characteristics, is Apache Hadoop. Within the Hadoop framework is the Hadoop Distributed File System (HDFS).
+HDFS is a distributed file system of the Hadoop technical framework that is developed based on the Google File System (GFS) and is used to manage files on multiple independent physical servers.
+
+It is applied in the following scenarios:
+-	Ecosystem data storage.
+-	Website user data behavior data storage.
+-	Meteorological data storage.
+
+### 2)	General Design of HDFS architecture
+The HDFS has design features of its architecture that enable its efficient working among which are the following;
+
+***Federation storage:*** HDFS creates a distinction between the namespace and storage. The two are separated to create a block storage layer.
+
+***High Availability:*** HDFS supports features such as data replication which enhances the availability of the system. A single block of data is replicated in 3 nodes so that should a single node fail, a client can access the block from 2 other nodes.
+Data can still be accessed normally even when a failure occurs on the DataNode or NameNode.
+
+***Multiple access modes:*** Within HDFS data can be accessed through HTTP on an HTTP browser, Java API for applications, or any other command shells.
+
+***Space reclamation:*** Space that had been released in HDFS can be reclaimed.
+This is implemented by a recycle bin mechanism where data that had been deleted can be restored from the recycle bin to occupy its initial space. The number of replicas can also be dynamically set.
+
+***NameNode/DataNode in master/slave mode:*** HDFS consists of NameNodes and DataNodes that work in a master/slave architecture. A single cluster consists of only one NameNode which regulates data access by clients and manages the namespace within the file system.
+
+The DataNode receives instructions from the NameNode on when to Create, Delete and replicate data blocks.
+
+***Unified file system Namespace:*** HDFS is presented externally as a coherent file system.
+Any external process perceives the system as one unified system.
+
+***Data replication:*** In HDFS, a file’s blocks are replicated for fault tolerance and the number of replicas can be specified by an application. This can be done at creation time but is subject to change at will.
+
+***Metadata persistence:*** The HDFS NameNode stores the namespace. The NameNode consistently records every change that occurs in file system metadata in a transaction log file called the EditLog. Whenever a new file is created in HDFS, the NameNode inserts a record into the EditLog indicating the creation of the new file.
```

Comment:
For `EditLog`, you could enclose them within ` `` `. Similarly, for any unique name or variable, you must enclose them within ` `` `.

Make this change wherever needed